medusa/search/manual.py

```diff
@@ -223,16 +222,30 @@ def get_provider_cache_results(indexer, show_all_results=None, perform_search=No
                     provider_id=cur_provider.get_id()
                 )
             )
-            additional_sql = " AND episodes LIKE ? AND season = ? "
 
-            # The params are always the same for both queries
+            # Let's start by adding the default parameters, which are used to subsitute the '?'s.
             add_params = [cur_provider.provider_type.title(), cur_provider.image_name(),
                           cur_provider.name, cur_provider.get_id(), minseed, minleech, show]
 
-            # If were not looking for all results, meaning don't do the filter on season + ep, add sql
-            if not int(show_all_results):
-                common_sql += additional_sql
-                add_params += ["%|{0}|%".format(sql_episode), season]
+            if manual_search_type != 'season':
+                # If were not looking for all results, meaning don't do the filter on season + ep, add sql
+                if not int(show_all_results):
+                    # If it's an episode search, pass season and episode.
+                    common_sql += " AND season = ? AND episodes LIKE ? "
+                    add_params += [season, "%|{0}|%".format(episode)]
+
+            else:
+                # If were not looking for all results, meaning don't do the filter on season + ep, add sql
+                if not int(show_all_results):
+                    list_of_episodes = '{0}{1}'.format(' episodes LIKE ', ' AND episodes LIKE '.join(
```

Comment:
This is the meat of this PR.
It essentially makes sure we query for a season pack when a season is passed and no episodes are passed. Which is the most often occurring situation. Or when all episodes for the season are passed. Like: |1|2|3|4|5|...etc|.